### ISSUE TITLE
wpt editor: Indicating NMPs

### ISSUE THREAD

**Issue by michihdeu** [TravelMapping/Web#393]
Once we can load the wpt file from Git, see #372 the next step would be indicating NMPs in the wpt editor. I suggest indicating this with a different back color similar to HDX: green, yellow and red.

This helps to fix NMPs and reduce the number of missing intersecting routes when navigating through HB.

**Comment by michihdeu** [comment]
Related but not a duplicate: #291

**Comment by yakra** [comment]
If the idea is to catch *new* errors as they're generated in real-time, reading tm-master.nmp or nearmisspoints.log is a no-go.
We don't have the siteupdate program's `WaypointQuadtree` structure in the DB, so this would be **incredibly SLOW.**

Even assuming a way to implement a WaypointQuadtree in the DB, that might not be enough.

Expensive queries. Multiple ones. For every point. Every time we click `load`, or edit a point in any way.
WPTedit is pretty lightweight and easy-to-use as it stands now. This could all change; it could become so bloated and bogged down with extra queries as to make it nigh unusable.

IMO, trying duplicate too much of datacheck's functionality, or that of the NMP files, into wptedit is the wrong road to go down.
The idea may be convenience, but we may end with the exact opposite of that.

**Comment by michihdeu** [comment]
No, I just thought about static data, just "loading" the nmp file from the last site update. It's more a "maintain active systems" feature, not a "develop new systems" thing.

Let's implement the first steps and come back to this later. If it won't be doable then, we can close it.

**Comment by jteresco** [comment]
Some of what was learned/implemented in the Scrollable Mapview might be helpful here as far as detecting new NMPs as routes are being created and edited.

**Comment by yakra** [comment]
All I know about working with SQL & webpages, I learned by getting under the hood of the various PHP files I've edited. Some of the wptedit issues got me thinking about using SQL queries in javascript, but I don't really know anything about that.
@jteresco, what's some of the stuff you've learned?
Do you have a handy link to any tutorials or how-tos, some starting points I can look at on the web?
And if the answer's not straightfoward & quick, not to worry. I can always Google `sql javascript` myself. :)

**Comment by jteresco** [comment]
> @jteresco, what's some of the stuff you've learned?
> Do you have a handy link to any tutorials or how-tos, some starting points I can look at on the web?

The idea's not too bad.  See the ajax calls in places like

https://github.com/TravelMapping/Web/blob/39c11d7f2cba15b0062839ccae4aafc11b5dbe55/lib/mapviewfuncs.js#L330-L336

and the corresponding PHP files in lib, and then the handler code back in JS to interpret and use the results.  In the case of the above, the PHP file in 

https://github.com/TravelMapping/Web/blob/master/lib/getRegionSystemSegments.php

gets the data, and the function starting here

https://github.com/TravelMapping/Web/blob/39c11d7f2cba15b0062839ccae4aafc11b5dbe55/lib/mapviewfuncs.js#L382 

parses the response.

**Comment by yakra** [comment]
Nice. I have 0 familiarity with ajax; that'll be something to learn about.
